### DISCUSSION TITLE
Credentials provider pass error message to client?

### DISCUSSION THREAD

**Discussion by nemanjam** [nextauthjs/next-auth#3972] [Help]
I have credentials provider and I want to pass "User does not exist" and "Incorrect password" to the client but even when I throw custom error object from `authorize` callback with that message I get `status: 200` and `response.ok = true`. If I return null I get `status: 401` but can not pass error message to the client obviously.

Why is `authorize()` so non intuitive and returns 200 with `throw`? How to pass error message with appropriate status code?

```typescript
CredentialsProvider({
  name: 'Credentials',
  credentials: {
	email: {
	  label: 'Email',
	  type: 'email',
	},
	password: {
	  label: 'Password',
	  type: 'password',
	},
  },
  async authorize(credentials) {
	const { user, error } = await getUser(credentials);
	if (error) throw error; // 200 ok, why?
	return user;
  },
}),
```

![image](https://user-images.githubusercontent.com/9990165/153855948-94d41077-b44a-40f8-8b56-1078af893033.png)

**Comment by veewoo** [comment]
Hi, I've just found this answer on the StackOverflow

https://stackoverflow.com/questions/70897330/return-error-information-from-api-when-using-next-auth

**Reply by nemanjam** [reply]
This will return 200 and not 401.

**Reply by veewoo** [reply]
Did you try "throw new Error(error)" instead of "throw error"?

**Reply by nemanjam** [reply]
error is of type ApiError which extends Error.

![image](https://user-images.githubusercontent.com/9990165/182204257-d7d323d7-ab4d-4108-bd0e-c6c2f1c2411e.png)

**Comment by alex-statsig** [comment]
@nemanjam Did you find a cleaner way to handle this? Running into the exact same problem, have basically accepted that a failed auth will give status 200 so I can pass an error message

**Reply by THE-SIMPLE-MARK** [reply]
You could perhaps throw an error, which next-auth then sends to your specified error page as a query parameter, [as described here](https://next-auth.js.org/providers/credentials#example---username--password). Then you could [set said error page to your login page](https://next-auth.js.org/configuration/pages), so now you are left with an error query parameter of your choice on your login page. You can then use this how you want on your frontend.

**Reply by dellwatson** [reply]
@THE-SIMPLE-MARK but i dont want to send it to another page when it was just error of invalid password or something, the error message can be shown under the field. 

also error page means, for both regis and signup will redirect to same page.

**Reply by THE-SIMPLE-MARK** [reply]
@dellwatson As I said, I solved that by setting my error page as my login page. This way you stay on the page and is only refreshed to load the new query parameter. Then you can put it under whatever field you want. The only slight inconvenience I can see is that the page refreshes when an error is returned.

**Comment by iamifechi** [comment]
This helped me resolve the issue 

https://stackoverflow.com/questions/70897330/return-error-information-from-api-when-using-next-auth
